### PR TITLE
Split AWS Config modules

### PR DESCRIPTION
**Pull request by ryansb**
##### SUMMARY

Takes the single big module proposed by @slapula https://github.com/ansible/ansible/pull/39080 and pulls them into smaller modules for easier docs/usage. 

These are all in one PR for now, but I can pull them apart

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

AWS Config

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by slapula** [comment]
Thanks @ryansb. As a general rule, is the preferred pattern here to err on the side of keeping modules small and specific vs large and comprehensive?  I actually have an AWS Storage Gateway module that's pretty much done but it's very much like my large AWS Config module.  If keeping modules small is preferred then I'll go ahead and break that one up too before I submit it.

**Comment by ryansb** [comment]
@slapula Yeah, generally we prefer to go small and then let the modules grow with the services. We used to do bigger modules (like `ec2` or `ec2_vpc`) but the usability suffered as those services added settings and features. Small modules give us a better way to grow over time, and makes documenting/using them easier. The larger config module was extremely nicely written - pulling it into separate ones turned out not to be too invasive - thanks for contributing it with the integration tests.

File: test/integration/targets/aws_config/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    # ============================================================
+    - name: Destroy Configuration Recorder
+      aws_config_recorder:
+        <<: *aws_connection_info
+        name: test_configuration_recorder
+        state: absent
+      register: output
+      ignore_errors: yes
+
+#    - assert:
```
Comment (s-hertel) [inline test/integration/targets/aws_config/tasks/main.yaml:334]: The commented out assertions in the `always` could be removed.

**Review by s-hertel — APPROVED** [review]
This looks great.

**Comment by s-hertel** [comment]
bot_status

**Comment by s-hertel** [comment]
!needs_revision

**Comment by mattclay** [comment]
@ryansb Why is the `aws_config` test marked `disabled`?

It looks like it probably should be `unsupported` instead?

**Pull request merged**